lib/rules/no-restricted-modules.js

```diff
@@ -119,14 +128,31 @@ module.exports = {
             return node.callee.type === "Identifier" && node.callee.name === "require";
         }
 
+        /**
+         * Extract string from Literal or TemplateLiteral node
+         * @param {ASTNode} node The node to extract from
+         * @returns {string|null} Extracted string or null if node doesn't represent a string
+         */
+        function getFirstArgumentString(node) {
+            if (isStringLiteral(node)) {
+                return node.value.trim();
+            }
+
+            if (isStringTemplateLiteral(node)) {
+                return node.quasis[0].value.raw.trim();
+            }
```

Comment:
This could produce a false negative on:

```js
/* eslint no-restricted-modules: ["error", "foo"] */

require(`fo\o`);
```

It should be `value.cooked` instead of `value.raw`.

Reply:
I'll also add fixture for it I guess

Reply:
> I'll also add fixture for it I guess

That would be great! And for the other note (with expressions), too :)

Reply:
Welp, it is causing `no-useless-escape` in test file. For now I'll just commit with `// eslint-disable-next-line no-useless-escape` so it's not forgotten, but I don't see other cases like that in test fixtures so it feels dirty to do so